packages/gestalt/src/PageHeaderComponents.js

```diff
@@ -0,0 +1,212 @@
+  hasThumbnail,
+  title,
+}: {|
+  hasThumbnail: boolean,
+  title: string,
+|}): Node {
+  return (
+    <Fragment>
+      <Box display="block" smDisplay="none">
+        <Heading size="400" lineClamp={1} accessibilityLevel={1}>
+          {title}
+        </Heading>
+      </Box>
+      <Box
+        display="none"
+        smDisplay="block"
+        dangerouslySetInlineStyle={{ __style: hasThumbnail ? { marginTop: '-5px' } : {} }}
+      >
+        <Heading size="500" lineClamp={1} accessibilityLevel={1}>
+          {title}
+        </Heading>
+      </Box>
+    </Fragment>
+  );
+}
+
+export function PageHeaderThumbnail({ thumbnail }: {| thumbnail: Element<typeof Image> |}): Node {
+  return (
+    <Box display="none" smDisplay="block" aria-hidden>
+      <Mask height={48} rounding={2} width={48}>
+        {thumbnail}
+      </Mask>
+    </Box>
+  );
+}
+
+export function PageHeaderBadge({
+  badgeText,
+  badgeTooltipText,
+}: {|
+  badgeText: string,
+  badgeTooltipText?: string,
+|}): Node {
+  return (
+    <Fragment>
+      {badgeText && !badgeTooltipText ? (
+        <Badge text={badgeText} type="info" position="top" />
+      ) : null}
+      {badgeText && badgeTooltipText ? (
+        <Tooltip accessibilityLabel="" text={badgeTooltipText} idealDirection="up">
+          <Badge text={badgeText} type="info" position="top" />
+        </Tooltip>
+      ) : null}
+    </Fragment>
+  );
+}
+
+export function PageHeaderHelperIconButton({
+  accessibilityLabel,
+  accessibilityControls,
+  accessibilityExpanded,
+  onClick,
+}: {|
+  accessibilityLabel: string,
+  accessibilityControls: string,
+  accessibilityExpanded: boolean,
+  onClick: ({|
+    event:
+      | SyntheticMouseEvent<HTMLButtonElement>
+      | SyntheticKeyboardEvent<HTMLButtonElement>
+      | SyntheticMouseEvent<HTMLAnchorElement>
+      | SyntheticKeyboardEvent<HTMLAnchorElement>,
+    dangerouslyDisableOnNavigation: () => void,
+  |}) => void,
+|}): Node {
+  return (
+    <Box dangerouslySetInlineStyle={{ __style: { marginTop: '-1px' } }} rounding="circle">
+      <IconButton
+        accessibilityControls={accessibilityControls}
+        accessibilityExpanded={accessibilityExpanded}
+        accessibilityHaspopup
+        accessibilityLabel={accessibilityLabel}
+        bgColor="lightGray"
+        icon="question-mark"
+        iconColor="darkGray"
+        onClick={onClick}
+        size="xs"
+      />
+    </Box>
+  );
+}
+
+export function PageHeaderSubtext({
+  subtext,
+  helperLink,
+}: {|
+  subtext: string,
+  helperLink?: {|
+    text: string,
+    accessibilityLabel: string,
+    href: string,
+    onClick: ({|
+      event: SyntheticMouseEvent<HTMLAnchorElement> | SyntheticKeyboardEvent<HTMLAnchorElement>,
+      dangerouslyDisableOnNavigation: () => void,
+    |}) => void,
+  |},
+|}): Node {
+  return (
+    <Box display="none" smDisplay="block">
+      <Text inline>
+        {subtext}
+        {helperLink ? (
+          <Fragment>
+            {' '}
```

Comment:
Did you try using Flex with `<Flex.Item flex="none">` around this block to allow using `gap` and avoid this hard-coded whitespace?

Reply:
![Screen Shot 2022-04-04 at 3 50 23 PM](https://user-images.githubusercontent.com/10593890/161620920-d41b9878-9ce7-49a1-b399-dc0ebea0dedb.png)

Reply:
Can you be more specific about what isn't matching the design spec here? It's difficult for me to offer suggestions based on an out-of-context screenshot